opm/input/eclipse/Schedule/ScheduleState.hpp

```diff
@@ -385,44 +385,7 @@ namespace Opm {
 
         template <typename T>
         ptr_member<T>& get() {
-            if constexpr ( std::is_same_v<T, PAvg> )
-                             return this->pavg;
-            else if constexpr ( std::is_same_v<T, WellTestConfig> )
-                                  return this->wtest_config;
-            else if constexpr ( std::is_same_v<T, GConSale> )
-                                  return this->gconsale;
-            else if constexpr ( std::is_same_v<T, GConSump> )
-                                  return this->gconsump;
-            else if constexpr ( std::is_same_v<T, GroupEconProductionLimits> )
-                                  return this->gecon;
-            else if constexpr ( std::is_same_v<T, WListManager> )
-                                  return this->wlist_manager;
-            else if constexpr ( std::is_same_v<T, Network::ExtNetwork> )
-                                  return this->network;
-            else if constexpr ( std::is_same_v<T, Network::Balance> )
-                                  return this->network_balance;
-            else if constexpr ( std::is_same_v<T, RPTConfig> )
-                                  return this->rpt_config;
-            else if constexpr ( std::is_same_v<T, Action::Actions> )
-                                  return this->actions;
-            else if constexpr ( std::is_same_v<T, UDQActive> )
-                                  return this->udq_active;
-            else if constexpr ( std::is_same_v<T, NameOrder> )
-                                  return this->well_order;
-            else if constexpr ( std::is_same_v<T, GroupOrder> )
-                                  return this->group_order;
-            else if constexpr ( std::is_same_v<T, UDQConfig> )
-                                  return this->udq;
-            else if constexpr ( std::is_same_v<T, GasLiftOpt> )
-                                  return this->glo;
-            else if constexpr ( std::is_same_v<T, GuideRateConfig> )
-                                  return this->guide_rate;
-            else if constexpr ( std::is_same_v<T, RFTConfig> )
-                                  return this->rft_config;
-            else if constexpr ( std::is_same_v<T, RSTConfig> )
-                                  return this->rst_config;
-            else
-                static_assert(always_false1<T>::value, "Template type <T> not supported in get()");
+            return const_cast<ptr_member<T>&>(std::as_const(*this).get<T>());
```

Comment:
Do you need `.template get<T>()` here?

Reply:
gcc is happy but clang might insist so better safe than sorry.